File: server/google-login/src/main/java/com/example/googlelogin/web/IndexController.java
package com.example.googlelogin.web;

import com.example.googlelogin.config.auth.LoginUser;
import com.example.googlelogin.config.auth.dto.SessionUser;
import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Controller;
import org.springframework.ui.Model;
import org.springframework.web.bind.annotation.GetMapping;

@RequiredArgsConstructor
@Controller
public class IndexController {

    @GetMapping("/")
    public String index(Model model, @LoginUser SessionUser user) {

        if (user != null) {
            model.addAttribute("userName", user.getName());
        }
        return "index";
    }

}

File: server/stock-api/src/main/kotlin/study/kstock/stockapi/web/IndexController.kt
package study.kstock.stockapi.web

import org.springframework.stereotype.Controller
import org.springframework.ui.Model
import org.springframework.web.bind.annotation.GetMapping
import study.kstock.stockapi.config.auth.dto.SessionUser
import javax.servlet.http.HttpSession

@Controller
class IndexController(private var httpSession: HttpSession) {

    @GetMapping("/")
    fun index(model: Model): String {
        var user = httpSession.getAttribute("user")
        if (user != null) {
            user = user as SessionUser
            model.addAttribute("userName", user.name)
        }
        return "index"
    }
}